include/range/v3/view/transform.hpp

```diff
@@ -287,13 +288,17 @@ namespace ranges
                 SizedRange<Rng1 const>() && SizedRange<Rng2 const>())
             constexpr size_type_ size() const
             {
-                return std::min<size_type_>(ranges::size(rng1_), ranges::size(rng2_));
+                return ranges::on(ranges::min,
+                    ranges::compose(convert_to<size_type_>{}, ranges::size))(
+                        rng1_, rng2_);
```

Comment:
Would you prefer:
```c++
auto f = compose(convert_to<size_type_>{}, ranges::size);
return ranges::min(f(rng1_), f(rng2_));
```
or
```c++
return ranges::min(
    static_cast<size_type_>(ranges::size(rng1_)),
    static_cast<size_type_>(ranges::size(rng2_)));
```
It's a shame that we can't have both `ranges::min(x, y)` and `ranges::min<T>(x, y)` be valid syntax when `ranges::min` is an object. I wonder if anyone has poked EWG about making:
```c++
template <class> class some_class_template;
template<class T = int>
some_class_template<T> foo{};

auto* x = &foo;
```
well-formed.